lib/app/handler/handler.go

```diff
@@ -823,15 +823,45 @@ func (h *WebHandler) telekubeInstallScript(w http.ResponseWriter, r *http.Reques
 		ver = constants.LatestVersion
 	}
 
-	// make sure the provided version is a valid semver
+	// Check whether the terraform provider is available
+	// Note: The terraform provider for older releases may not be published, so the following code tries to detect
+	// whether the terraform provider has been published for the requested release
+	//
+	// This works by attempting to locate the package, and resolving the version in the tfVersion variable:
+	// NotFound  -> ""          - Don't try and install if it doesn't exist for the specified version
+	// <version> -> "<version>" - If a specific version is requested, install that version
+	// latest    -> "<version>" - If latest is specified, resolve that to the latest available version.
+	tfVersion := pack.LatestVersion
```

Comment:
nit: We already have `constants.LatestVersion` used above - I think one constant should suffice.

Reply:
Also, I think it would be better to factor the code that determines the version of terraform plugin into a separate function:
```go
func (h *WebHandler) telekubeInstallScript(w http.ResponseWriter, r *http.Request, p httprouter.Params) error {
	ver := p.ByName("version")
	if ver == "" {
		ver = constants.LatestVersion
	}

	tfVersion, err := getTerraformVersion(ver, h.Packages)
	if err != nil {
		return trace.Wrap(err)
	}

	err = telekubeInstallScriptTemplate.Execute(w, map[string]string{
		"version":   ver,
		"tfVersion": tfVersion,
	})
	if err != nil {
		return trace.Wrap(err)
	}

	return nil
}

func getTerraformVersion(binaryVersion string, packages pack.PackageService) (tfVersion string, err error) {
	// Check whether the terraform provider is available
	// Note: The terraform provider for older releases may not be published, so the following code tries to detect
	// whether the terraform provider has been published for the requested release
	//
	// This works by attempting to locate the package, and resolving the version in the tfVersion variable:
	// NotFound  -> ""          - Don't try and install if it doesn't exist for the specified version
	// <version> -> "<version>" - If a specific version is requested, install that version
	// latest    -> "<version>" - If latest is specified, resolve that to the latest available version.
	tfVersion = constants.LatestVersion
	if ver != constants.LatestVersion {
		_, err := semver.NewVersion(binaryVersion)
		if err != nil {
			return "", trace.BadParameter("the provided version is not valid: %v", ver)
		}
		tfVersion = binaryVersion
	}

	// hard code our module lookup based on linux/x86_64, if it exists, we assume it exists
	// for other requested architectures/os, which won't be detected until later
	name := strings.Join([]string{constants.TerraformGravityPackage, "linux", "x86_64"}, "_")
	locator, err := loc.NewLocator(defaults.SystemAccountOrg, name, tfVersion)
	if err != nil {
		return "", trace.Wrap(err)
	}

	envelope, err := .ReadPackageEnvelope(*locator)
	if err != nil {
		if !trace.IsNotFound(err) {
			return "", trace.Wrap(err)
		}
		tfVersion = ""
	} else {
		tfVersion = envelope.Locator.Version
	}

	return tfVersion, nil
}
```

Reply:
pack.LatestVersion / constants.LatestVersion are two different values unfortunately. constants.LatestVersion = "latest", pack.LatestVersion = "0.0.0+latest". So I had to follow the convention already in use.